src/libraries/Microsoft.PowerFx.LanguageServerProtocol/LanguageServer/LanguageServer.cs

```diff
@@ -475,6 +479,24 @@ private void PublishTokens(string documentUri, CheckResult result)
                 }));
         }
 
+        private void PublishExpressionType(string documentUri, CheckResult result)
+        {
+            var uri = new Uri(documentUri);
+            var nameValueCollection = HttpUtility.ParseQueryString(uri.Query);
+            if (!bool.TryParse(nameValueCollection.Get("getExpressionType"), out var enabled) || !enabled)
+            {
+                return;
+            }
```

Comment:
This notification is fired on every change notification (along with other error notifications)? Hence the need for a switch (to avoid bombarding old clients).

Reply:
Exactly my thinking. Was this your response to your own question above?